survival_model_simulator.py

```diff
@@ -1,29 +1,38 @@
 import numpy as np
 from common import mutate_string
 from common import NUCLEOTIDES, NUM_NUCLEOTIDES
-from common import sample_multinomial
+from common import sample_multinomial, process_degenerates_and_impute_nucleotides
 from models import *
 
 class SurvivalModelSimulator:
     """
     A simple model that will mutate sequences based on the survival model we've assumed.
     We will suppose that the hazard is constant over time.
     """
-    def simulate(self, start_seq, censoring_time=None, percent_mutated=None, with_replacement=False):
+    def simulate(self, start_seq=None, censoring_time=None, percent_mutated=None, with_replacement=False, obs_seq_mutation=None):
```

Comment:
I think it would be cleaner if the arguments were `start_seq`, `left_flank`, `right_flank` instead of `start_seq` or `obs_seq_mutation`.

I'm imagining you would call this function as
`simulate(obs_seq_mutation.start_seq,obs_seq_mutation.left_flank, obs_seq_mutation.right_flank,...)`
or
`simulate(start_seq,None, None,...)`